### PR TITLE
PrestaShop 8 Compatibility

### PR DESCRIPTION
**Pull request by jeckyl**

### CODE REVIEW

**Review by yaloumi — APPROVED** [review]

File: src/services/VersionValidatorService.php [hunk 1]
```diff
@@ -58,7 +58,7 @@ public function getActualVersionData()
             ));
 
             $releaseJsonData = Tools::file_get_contents(self::PLUGIN_RELEASE_DATA_URL, false, $context);
-            $releaseData = Tools::jsonDecode($releaseJsonData, true);
+            $releaseData = json_decode($releaseJsonData, true);
```
Comment (rkistaps) [inline src/services/VersionValidatorService.php:61]: Isn't it required to use the `Tools` class anymore?

**Comment by flaviordrgo** [comment]
Pls 8.1 Compatibility

File: src/services/VersionValidatorService.php (hunk 1 above)
Comment (nickoe) [reply src/services/VersionValidatorService.php:61]: I don't know php at all, but all I know is that using `Tools::jsonDecode` does not work for me on prestashop 8 and php 8.1, whereas `json_decode` appears to not cause internal server errors for prestashop. I don't know if that is the right method to use at all, I think I also see this when grepping for json_decode...

```
endor/guzzlehttp/guzzle/src/Utils.php:    public static function jsonDecode(string $json, bool $assoc = false, int $depth = 512, int $options = 0)
vendor/guzzlehttp/guzzle/src/functions.php: * @deprecated json_decode will be removed in guzzlehttp/guzzle:8.0. Use Utils::jsonDecode instead.
vendor/guzzlehttp/guzzle/src/functions.php:    return Utils::jsonDecode($json, $assoc, $depth, $options);
```

Maybe `Utils::jsonDecode` is the right thing?

I don't know how you test the plugin currently, but maybe you should try out the binami docker image?

Please see:
https://github.com/bitnami/containers/tree/main/bitnami/prestashop#how-to-use-this-image

----
The error I am getting is:
```
var/logs/prod.log:[2023-04-18 10:57:34] request.CRITICAL: Uncaught PHP Exception Symfony\Component\ErrorHandler\Error\UndefinedMethodError: "Attempted to call an undefined method named "jsonDecode" of class "Tools"." at /bitnami/prestashop/modules/printful/src/services/VersionValidatorService.php line 61 {"exception":"[object] (Symfony\\Component\\ErrorHandler\\Error\\UndefinedMethodError(code: 0): Attempted to call an undefined method named \"jsonDecode\" of class \"Tools\". at /bitnami/prestashop/modules/printful/src/services/VersionValidatorService.php:61)"} []
```

Comment (nickoe) [reply src/services/VersionValidatorService.php:61]: @rkistaps It appears it was removed in prestashop here
https://github.com/PrestaShop/PrestaShop/commit/c1d84724067c6fecf75745b9f7eb8d419fe9002d

**Review by nickoe — APPROVED** [review]

File: src/services/VersionValidatorService.php (hunk 1 above)
Comment (IvanMtzBoerne) [reply src/services/VersionValidatorService.php:61]: I just commented on the version verification and works well

**Comment by ddw-0** [comment]
Experiencing incompatibility issues with Printful module on PrestaShop 8.1.1 and PHP 8.1. Unfortunately, this has led to disruptions in store operations. Hoping for a swift resolution from the Printful team.

**Comment by Rapsssito** [comment]
@rkistaps, any possibility this could be merged?

**Comment by rkistaps** [comment]
Hello. We have increased the compatibility of our plugin with the latest PrestaShop version 8.1.3. Please use Printful Prestashop module v2.0 for the newest PrestaShop.

**Comment by orlandopadillad** [comment]
I have the following issue trying to install the 8.1.5

Cannot Install module printful. The version of your module is not compliant with your PrestaShop version.

**Pull request closed without merging**